### PR TITLE
fix: merge and pull API issues

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description

> Change the request params to be passed via query params

Fixes # (issue)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Locally via Postman
- JUnit Tests

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/dtos/MergeStatusDTO.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 @Getter
 @Setter
 public class MergeStatusDTO {
-    boolean isMerge;
+    boolean isMergeAble;
```
Comment (abhvsn) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/dtos/MergeStatusDTO.java:11]: Nit : Can we use isMergeable?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java [hunk 2]
```diff
@@ -803,27 +803,32 @@ private String getRepoName(String remoteUrl) {
                                 if (error.getMessage().contains("Nothing to fetch.")) {
                                     MergeStatusDTO mergeStatus = new MergeStatusDTO();
                                     mergeStatus.setStatus("Nothing to fetch from remote. All changes are upto date.");
-                                    mergeStatus.setMerge(false);
+                                    mergeStatus.setMergeAble(false);
                                     return Mono.just(mergeStatus);
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java:807]: Shouldn't we stop the pull flow here as there is nothing to pull in remote

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java [hunk 3]
```diff
@@ -803,27 +803,32 @@ private String getRepoName(String remoteUrl) {
...
-                                    MultiValueMap<String, String> valueMap = new LinkedMultiValueMap<>();
-                                    valueMap.add(FieldName.BRANCH_NAME, branchName);
-                                    return handleMergeConflict(defaultApplicationId, valueMap)
-                                            .flatMap(status -> Mono.error(new AppsmithException(AppsmithError.GIT_ACTION_FAILED, "pull",error.getMessage() )));
+                                    // On merge conflict send the response with the error message
+                                    MergeStatusDTO mergeStatus = new MergeStatusDTO();
+                                    mergeStatus.setStatus(error.getMessage());
+                                    mergeStatus.setMergeAble(false);
+                                    return Mono.just(mergeStatus);
+                                    //return Mono.error(new AppsmithException(AppsmithError.GIT_ACTION_FAILED, " pull", error.getMessage()));
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java:814]: Let's return error here with the available error message and force reset the repo to last commit

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java (hunk 2 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java:807]: This is to send the FE that merge failed and list of conflicting files. As per the 3 states design change for error state

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java (hunk 3 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java:814]: This is to send the FE that merge failed and list of conflicting files. As per the 3 states design change for error state.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java (hunk 2 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java:807]: Yeah my bad, i will change this.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java (hunk 3 above)
Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/GitServiceImpl.java:814]: I didn't get why we are chaining as we are just doing the import app from repo which is not required as all the changes are already present in DB considering there are merge conflicts

**Review by abhvsn — COMMENTED** [review]
As we are not making any decisions on the BE about creating a new branch when app is in conflicts state can you please remove `handleMergeConflicts` method

**Pull request closed without merging**